app/assets/javascripts/discourse/components/topic-list-item.js.es6

```diff
@@ -54,6 +55,19 @@ export default Ember.Component.extend(StringBuffer, {
     return classes.join(' ');
   }.property(),
 
+  lastVisitMessage: function() {
+    if (this.get('lastVisitedTopic') == this.get('topic')) {
+      this.session.set('lastVisitedTopicAbove', true);
```

Comment:
all this session stuff is really strange, its a computed property ... 

also ... I don't think this would work properly if you drilled into a category cause the line can move.